### PR TITLE
Refactored some of the code to support disabling of analyses

### PR DESCRIPTION
**Pull request by jcomack**
This applies to both content and SEO analysis.

This means that things like Assessors, Presenters etc. don't get loaded unless these are marked as being active.

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -264,6 +254,46 @@ var App = function( args ) {
 	this.refresh();
 };
 
+App.prototype.getDefaultOutputElement = function( args ) {
+	if ( args.keywordAnalysisActive ) {
```
Comment (atimmer) [inline js/app.js]: You should add `keywordAnalysisActive` to the default arguments of the `App`.

File: spec/appSpec.js [hunk 2]
```diff
@@ -78,19 +78,6 @@ describe( "Creating an App", function(){
 		});
 	});
 
-	it( "throws on a missing output element ID", function() {
```
Comment (atimmer) [inline spec/appSpec.js]: When changing the behaviour you shouldn't delete the test case that tested this behaviour, instead you should change the expected behaviour of the test.

In this case the spec has changed from "throws on a missing output element ID" -> "works when no output element ID has been passed".

File: js/app.js [hunk 3]
```diff
@@ -264,6 +257,46 @@ var App = function( args ) {
...
+	}
+
+	if ( args.contentAnalysisActive ) {
+		return args.targets.contentOutput;
+	}
+
+	return "";
+};
+
+App.prototype.initializeAssessors = function( args ) {
```
Comment (terw-dan) [inline js/app.js]: Missing JSdoc

File: js/app.js [hunk 4]
```diff
@@ -264,6 +257,46 @@ var App = function( args ) {
...
+	if ( args.keywordAnalysisActive ) {
+		this.initializeSEOAssessor( args );
+	}
+
+	if ( args.contentAnalysisActive ) {
+		this.initializeContentAssessor( args );
+	}
+};
+
+App.prototype.initializeSEOAssessor = function( args ) {
```
Comment (terw-dan) [inline js/app.js]: Missing JSdoc

File: js/app.js [hunk 5]
```diff
@@ -264,6 +257,46 @@ var App = function( args ) {
...
+	// Set the assessor
+	if ( isUndefined( args.seoAssessor ) ) {
+		this.seoAssessor = new SEOAssessor( this.i18n, { marker: this.config.marker } );
+	} else {
+		this.seoAssessor = args.seoAssessor;
+	}
+};
+
+App.prototype.initializeContentAssessor = function( args ) {
```
Comment (terw-dan) [inline js/app.js]: Missing JSdoc

File: js/app.js [hunk 6]
```diff
@@ -264,6 +257,46 @@ var App = function( args ) {
 	this.refresh();
 };
 
+App.prototype.getDefaultOutputElement = function( args ) {
```
Comment (terw-dan) [inline js/app.js]: Missing JSdoc

**Comment by terw-dan** [comment]
CR Done

**Comment by terw-dan** [comment]
CR done :+1:

**Comment by boblinthorst** [comment]
Acceptance tests: mergeable with knowledge of some minor bugs

**Pull request merged**